tests/GridFS/ReadableStreamFunctionalTest.php

```diff
@@ -108,14 +110,34 @@ public function provideFileIdAndExpectedBytes()
         ];
     }
 
+    public function provideFilteredFileIdAndExpectedBytes()
```

Comment:
It looks like there is no way to chain data providers so I thought this would be the most straightforward way to provide data to `testReadBytesCalledMultipleTimes()`

Reply:
If you want to avoid duplicating the datasets from `provideFileIdAndExpectedBytes`, you could either call that data provider and run its return value through `array_filter`. Altternatively, `testReadBytesCalledMultipleTimes` could just skip all datasets that don't apply with `markTestSkipped`. Not sure if any of those sound better to you, just providing some ideas.

Reply:
> It looks like there is no way to chain data providers

Were you by chance reading about [`@depends`](https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.depends)? That's the only form of test chaining I'm aware of, where the return value of one test can be used as input for a subsequent test case.

Anywho, I think I mentioned using [`array_filter()`](php.net/array_filter) earlier this week, which is exactly what @alcaeus is proposing here. Since `provideFileIdAndExpectedBytes()` returns an array of argument arrays, and we just want to filter out any argument arrays where the length (i.e. second argument) is zero by using an anonymous function as our filter callback:

```php
function(array $args) { return $args[1] > 0; }
```

Using `provideFileIdAndExpectedBytes()` as the `array_filter()` input, each of its 20 argument arrays will be run through our callback and only those where the callback returns `true` will be kept. This allows us to re-use most of the existing data without duplicating it in the test sources.